staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -691,6 +691,30 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 	}
 }
 
+// requestPreflightCheck looks for common programmer errors on Request.
+//
+// We return an error if an attempt to POST is made using an "empty" namespace
```

Comment:
We have cluster scoped resources like Nodes, ClusterRoles, and many others.  This comment suggests that this change would prevent those from working.

Reply:
Well, maybe the comment is not clear enough. What I meant is that if `r.namespaceSet == true` then `len(r.namespace)` can be zero(empty). This logic was already there, but in a different way[1].


[1] https://github.com/kubernetes/kubernetes/pull/82652/files#diff-a47b7155403b5094bbe75cf02d4da303L710

Reply:
Absolutely. It is done, ptal.